project_tracker/mock_pub.py

```diff
@@ -0,0 +1,69 @@
+import rclpy
+from rclpy.node import Node
+
+import numpy as np
+
+from std_msgs.msg import Header
+from sensor_msgs.msg import PointCloud2 as PCL2
+from sensor_msgs.msg import PointField
+from sensor_msgs_py import point_cloud2
+
+class PointCloudToPCL2(Node):
+
+    def __init__(self):
+        super(PointCloudToPCL2, self).__init__('point_cloud_to_pcl2')
+        self._publisher = self.create_publisher(PCL2, 'pcl2conversion', 10)
+
+        # TODO -> figure out way to have this callback executed when next .bin file
+        #         is available and not on a timer
+        timer_period = 0.5
+        self.timer = self.create_timer(timer_period, self.publish_pcl2)
+        self.velodyne_file_path = '/home/benca/mcav-dev/src/project_tracker/data/2011_09_26/2011_09_26_drive_0048_sync/velodyne_points/data/0000000000.bin'
```

Comment:
### About the TODO

I see that you are loading a single .bin file, waiting for the timer and then publishing it. Why don't you load all bin directories using `glob` and then publish use `self.publish_pcl2()` to publish each new one every `timer_callback`?

You can add an indexing variable to `__init__()` and load all `.bin` files using:
```python
import glob

# in the PointCloudToPCL2 class 
    def __init__(self):
        super(PointCloudToPCL2, self).__init__('point_cloud_to_pcl2')
        self._publisher = self.create_publisher(PCL2, 'pcl2conversion', 10)

        timer_period = 0.5
        self.timer = self.create_timer(timer_period, self.publish_pcl2)
        self.velodyne_file_paths = glob.glob('/home/benca/mcav-dev/src/project_tracker/data/2011_09_26/2011_09_26_drive_0048_sync/velodyne_points/data/*.bin')

        self.indx = 0
```
Now all you have to in `publish_pcl2()` is:
```python
    def publish_pcl2(self):
        """Callback to publish"""
        msg = self.convert_bin_to_PCL2(self.velodyne_file_paths(self.indx)) 
        self._publisher.publish(msg)

        self.indx += 1

```

Is this what you mean by that TODO? or is it something else?

Reply:
Thanks Laksh, yeah I believe that's what Ben meant